### PR TITLE
Reducing panic to error, with input for context

### PR DESCRIPTION
**Pull request by nyarly**

### CODE REVIEW

**Comment by sous-machine-account** [comment]
LGTM

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: util/restful/client.go [hunk 1]
```diff
@@ -311,7 +311,11 @@ func (client *LiveHTTPClient) buildRequest(method, url string, headers map[strin
 	if rqBody != nil {
 		JSON = encodeJSON(rqBody)
 		if resource != nil {
-			JSON = putbackJSON(resource.body, resource.resourceJSON, JSON)
+			var err error
+			JSON, err = putbackJSON(resource.body, resource.resourceJSON, JSON)
+			if err != nil {
+				return nil, err
```
Comment (JC1738) [inline util/restful/client.go]: We no longer log if err, does the caller log this error? @nyarly

Comment (nyarly) [reply util/restful/client.go]: Yes, I tracked the callchain, and the errors are logged - with more context.

**Comment by JC1738** [comment]
One of the builds past, merging.

**Pull request merged**